src/Spectre.Angular2Client/src/app/heatmap/heatmap.component.spec.ts

```diff
@@ -0,0 +1,66 @@
+/*
+ * heatmap.component.spec.ts
+ * Unit tests for single heatmap component.
+ *
+ Copyright 2017 Grzegorz Mrukwa, Sebastian Pustelnik, Daniel Babiak
+
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+ http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+ */
+
+import { async, ComponentFixture, TestBed } from '@angular/core/testing';
+import { Http, BaseRequestOptions, Response, ResponseOptions } from '@angular/http';
+import { MockBackend} from '@angular/http/testing';
+
+import { ActivatedRoute } from '@angular/router';
+import { RouterTestingModule } from '@angular/router/testing';
+
+import { PlotlyModule } from '../plotly/plotly.module';
+
+import { HeatmapService } from './heatmap.service';
+import {HeatmapComponent} from './heatmap.component';
+
+describe('HeatmapComponent', () => {
+  let component: HeatmapComponent;
+  let fixture: ComponentFixture<HeatmapComponent>;
+
+  beforeEach(async(() => {
+    TestBed.configureTestingModule({
+      declarations: [HeatmapComponent],
+      imports: [RouterTestingModule, PlotlyModule],
+      providers: [
+        MockBackend,
+        BaseRequestOptions,
+        {
+          provide: Http,
+          useFactory: (backendInstance: MockBackend, defaultOptions: BaseRequestOptions) => {
+            return new Http(backendInstance, defaultOptions);
+          },
+          deps: [MockBackend, BaseRequestOptions]
+        },
+        HeatmapService
+      ]
+    })
+      .compileComponents();
+  }));
+
+  beforeEach(() => {
+    fixture = TestBed.createComponent(HeatmapComponent);
+    component = fixture.componentInstance;
+    fixture.detectChanges();
+  });
+
+  it('should create', () => {
```

Comment:
I think some more tests should be welcome. At least, whether it fires updates on data change or do not crash on any data provided.